### PR TITLE
add trie for route

### PR DESCRIPTION
**Pull request by yqxu**
**What this PR does**: add trie struct for route request to matched filter

**Does this PR introduce a user-facing change?**: remove head match for request

### CODE REVIEW

File: pkg/common/http/manager_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ package http
 import (
 	"bytes"
 	"fmt"
+	"github.com/apache/dubbo-go-pixiu/pkg/common/router/trie"
```
Comment (AlexStocks) [inline pkg/common/http/manager_test.go:23]: guy, move it.

File: pkg/common/router/router.go [hunk 2]
```diff
@@ -18,6 +18,7 @@
 package router
 
 import (
+	"github.com/apache/dubbo-go-pixiu/pkg/common/constant"
```
Comment (AlexStocks) [inline pkg/common/router/router.go:21]: MOVE IT

File: pkg/common/router/router.go [hunk 3]
```diff
@@ -54,23 +55,50 @@ func (rm *RouterCoordinator) Route(hc *http.HttpContext) (*model.RouteAction, er
...
+func getTrieKey(method string, path string, isPrefix bool) string {
+	if isPrefix {
+		return method + constant.PathSlash + path + constant.PathSlash + "**"
+	}
+	return method + constant.PathSlash + path
+}
+
 // OnAddRouter add router
 func (rm *RouterCoordinator) OnAddRouter(r *model.Router) {
+	//锁粒度 下期优化到 字典树node 级别
```
Comment (AlexStocks) [inline pkg/common/router/router.go:67]: 第一，在此处加上 TODO
第二，不要使用中文注释

**Comment by AlexStocks** [comment]
IMPORT 使用规范，请教下其他同学。

代码中不要出现中文注释。

**Comment by mark4z** [comment]
Resolve conflicts pls

File: pkg/common/router/router.go [hunk 4]
```diff
@@ -54,23 +55,50 @@ func (rm *RouterCoordinator) Route(hc *http.HttpContext) (*model.RouteAction, er
...
+	isPrefix := r.Match.Prefix != ""
+	for _, method := range r.Match.Methods {
+		var key string
+		if isPrefix {
+			key = getTrieKey(method, r.Match.Prefix, isPrefix)
+		} else {
+			key = getTrieKey(method, r.Match.Path, isPrefix)
+		}
+		rm.activeConfig.RouteTrie.Put(key, r.Route)
+	}
```
Comment (mark4z) [inline pkg/common/router/router.go:96]: resolve it

File: pkg/common/constant/http.go [hunk 5]
```diff
@@ -43,3 +43,10 @@ const (
 	PprofDefaultAddress = "0.0.0.0"
 	PprofDefaultPort    = 7070
 )
+
+const (
+	Get    = "GET"
+	Put    = "PUT"
+	Post   = "POST"
+	Delete = "DELETE"
+)
```
Comment (MasterKenway) [inline pkg/common/constant/http.go:52]: What about using constant defined in the net/http?

**Comment by mark4z** [comment]
The impots format problem can be solved with the tools

**Comment by mark4z** [comment]
> The impots format problem can be solved with the tools

https://github.com/dubbogo/tools

**Review by mark4z — APPROVED** [review]

**Pull request merged**